File: TutorialVB/Controllers/EventController.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Security.Cryptography
Imports System.Web.Mvc
Imports DayPilot.Web.Mvc.Json

Namespace TutorialVB.Controllers
    Public Class EventController
        Inherits Controller

        Private dc As New CalendarDataContext()

        Public Function Edit(ByVal id As String) As ActionResult
            Dim ids = Convert.ToInt32(id)
            Dim t = ( _
                From tr In dc.Events _
                Where tr.Id = ids _
                Select tr).First()
            Dim ev = New EventData With {.Id = t.Id, .Start = t.Start, .End = t.End, .Text = t.Text}
            Return View(ev)
        End Function

        <AcceptVerbs(HttpVerbs.Post)> _
        Public Function Edit(ByVal form As FormCollection) As ActionResult
            Dim id As Integer = Convert.ToInt32(form("Id"))
            Dim start As Date = Convert.ToDateTime(form("Start"))
            Dim [end] As Date = Convert.ToDateTime(form("End"))
            Dim text As String = form("Text")

            Dim record = ( _
                From e In dc.Events _
                Where e.Id = id _
                Select e).First()
            record.Start = start
            record.End = [end]
            record.Text = text
            dc.SubmitChanges()

            Return JavaScript(SimpleJsonSerializer.Serialize("OK"))
        End Function


        Public Function Create() As ActionResult
            Return View(New EventData With {.Start = Convert.ToDateTime(Request.QueryString("start")), .End = Convert.ToDateTime(Request.QueryString("end"))})
        End Function

        <AcceptVerbs(HttpVerbs.Post)> _
        Public Function Create(ByVal form As FormCollection) As ActionResult
            Dim start As Date = Convert.ToDateTime(form("Start"))
            Dim [end] As Date = Convert.ToDateTime(form("End"))
            Dim text As String = form("Text")
            Dim resource As Integer = Convert.ToInt32(form("Resource"))
            'string recurrence = form["Recurrence"];

            Dim toBeCreated = New [Event]() With {.Start = start, .End = [end], .Text = text}
            dc.Events.InsertOnSubmit(toBeCreated)
            dc.SubmitChanges()

            Return JavaScript(SimpleJsonSerializer.Serialize("OK"))
        End Function

        Public Class EventData
            Public Property Id() As Integer
            Public Property Start() As Date
            Public Property [End]() As Date
            Public Property Resource() As SelectList
            Public Property Text() As String
        End Class

    End Class
End Namespace

File: TutorialCS/Controllers/EventController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Security.Cryptography;
using System.Web.Mvc;
using DayPilot.Web.Mvc.Json;
using TutorialCS.Models;

namespace TutorialCS.Controllers
{
    public class EventController : Controller
    {
        CalendarDataContext dc = new CalendarDataContext();

        public ActionResult Edit(string id)
        {
            var ids = Convert.ToInt32(id);
            var t = (from tr in dc.Events where tr.Id == ids select tr).First();
            var ev = new EventData
            {
                Id = t.Id,
                Start = t.Start,
                End = t.End,
                Text = t.Text
            };
            return View(ev);
        }

        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Edit(FormCollection form)
        {
            int id = Convert.ToInt32(form["Id"]);
            DateTime start = Convert.ToDateTime(form["Start"]);
            DateTime end = Convert.ToDateTime(form["End"]);
            string text = form["Text"];

            var record = (from e in dc.Events where e.Id == id select e).First();
            record.Start = start;
            record.End = end;
            record.Text = text;
            dc.SubmitChanges();

            return JavaScript(SimpleJsonSerializer.Serialize("OK"));
        }


        public ActionResult Create()
        {
            return View(new EventData
            {
                Start = Convert.ToDateTime(Request.QueryString["start"]),
                End = Convert.ToDateTime(Request.QueryString["end"])
            });
        }

        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Create(FormCollection form)
        {
            DateTime start = Convert.ToDateTime(form["Start"]);
            DateTime end = Convert.ToDateTime(form["End"]);
            string text = form["Text"];
            int resource = Convert.ToInt32(form["Resource"]);
            //string recurrence = form["Recurrence"];

            var toBeCreated = new Event() { Start = start, End = end, Text = text};
            dc.Events.InsertOnSubmit(toBeCreated);
            dc.SubmitChanges();

            return JavaScript(SimpleJsonSerializer.Serialize("OK"));
        }

        public class EventData
        {
            public int Id { get; set; }
            public DateTime Start { get; set; }
            public DateTime End { get; set; }
            public SelectList Resource { get; set; }
            public string Text { get; set; }
        }

    }
}
